File: src/main/java/myLang/block/statement/ReturnStatementContext.java
package myLang.block.statement;

import myLang.ContextHandler;
import myLang.expression.ExpressionContext;
import myLang.response.MyLangException;
import myLang.response.Response;
import myLang.response.ReturnResponse;
import myLangParser.MyLangParser;

public class ReturnStatementContext implements ContextHandler<MyLangParser.ReturnStatementContext> {
    @Override
    public Response handler(MyLangParser.ReturnStatementContext ctx) throws MyLangException {
        if(ctx.expression() != null){
            Response response = new ExpressionContext().handler(ctx.expression());
            return new ReturnResponse(String.valueOf(response.getResponse()));
        }
        return new ReturnResponse("");
    }
}

File: src/main/kotlin/myLangKotlin/block/statement/ReturnStatementContext.kt
package myLangKotlin.block.statement

import myLangKotlin.ContextHandler
import myLangKotlin.expression.ExpressionContext
import myLangKotlin.response.MyLangException
import myLangKotlin.response.Response
import myLangKotlin.response.ReturnResponse
import myLangParser.MyLangParser

class ReturnStatementContext : ContextHandler<MyLangParser.ReturnStatementContext> {
    @Throws(MyLangException::class)
    override fun handler(ctx: MyLangParser.ReturnStatementContext): Response<*> {
        if (ctx.expression() != null) {
            val response = ExpressionContext().handler(ctx.expression())
            return ReturnResponse(response.response.toString())
        }
        return ReturnResponse("")
    }
}
